client/crypt.c

```diff
@@ -300,7 +336,13 @@ static uint8_t *add_number_to_sign(uint8_t *sign, const xdag_hash_t num)
 	leadzero = (i < sizeof(xdag_hash_t) && n[i] & 0x80);
 	len = (sizeof(xdag_hash_t) - i) + leadzero;
 	*sign++ = 0x02;
-	*sign++ = len;
+	if(len)

```

Comment:
Can you clarify that condition please.

Reply:
the code of daniel doesn't follow the standard, but openssl was accepting it the same.
secp256k1 is more rigorous

Reply:
if the signature is 0....000 daniel code recognize it that length is 0, but length by DER is 1 when signature is 0
*sign++ = 1; <- length of the signature
*sign++ = 0; <- value of the signature
return sign;